rust/index/src/hnsw_provider.rs

```diff
@@ -143,8 +149,9 @@ impl HnswIndexProvider {
     pub async fn fork(
         &self,
         source_id: &Uuid,
-        segment: &Segment,
+        collection_id: &Uuid,
```

Comment:
technically this is leaky too

Reply:
I renamed it to cache_key. Not sure if there are alternatives